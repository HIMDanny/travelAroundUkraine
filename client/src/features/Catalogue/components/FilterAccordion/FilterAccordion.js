import React,{memo} from 'react';
import PropTypes from 'prop-types';
import { styled, Accordion, AccordionSummary, AccordionDetails, Typography } from '@mui/material';

import ExpandMoreIcon from '@mui/icons-material/ExpandMore';

const StyledAccordion = styled((props) => <Accordion elevation={0} square defaultExpanded {...props} />)(
  ({ theme }) => ({
    position: 'relative',
    boxShadow: 0,

    '&:after': {
      content: '""',
      position: 'absolute',
      bottom: 0,
      left: 0,
      height: '1px',
      width: '100%',
      borderTop: `1px solid ${theme.palette.divider}`,
    },

    '&:before': {
      background: 'none',
    }

  })
);

const StyledAccordionDetails = styled(AccordionDetails)({
  padding: '0 0 40px',
});

const AccordionTitle = styled(Typography)(() => ({
  padding: 0,
  flexShrink: 0,
  fontWeight: 700,
}));

const FilterAccordion = ({ title, children }) => (
  <StyledAccordion>
    <AccordionSummary sx={{ p: '0' }} expandIcon={<ExpandMoreIcon fontSize="medium" color="primary" />}>
      <AccordionTitle>{title}</AccordionTitle>
    </AccordionSummary>
    <StyledAccordionDetails>{children}</StyledAccordionDetails>
  </StyledAccordion>
);

FilterAccordion.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node,
};

FilterAccordion.defaultProps = {
  title: '',
  children: <p />,
};

export default memo(FilterAccordion);
