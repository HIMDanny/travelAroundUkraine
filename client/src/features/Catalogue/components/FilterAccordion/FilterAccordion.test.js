import { render } from '@testing-library/react';
import FilterAccordion from './FilterAccordion';

jest.mock('@mui/material/Accordion', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/AccordionSummary', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/AccordionDetails', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/Typography', () => ({ children }) => <p>{children}</p>);


describe('FilterAccordion snapshot test', () => {
  test('should FilterAccordion match snapshot', () => {
    const { asFragment } = render(<FilterAccordion />);
    expect(asFragment()).toMatchSnapshot();
  });
});
