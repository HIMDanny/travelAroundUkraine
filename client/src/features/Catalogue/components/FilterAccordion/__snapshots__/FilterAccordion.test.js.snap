// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterAccordion snapshot test should FilterAccordion match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <p />
    </div>
    <div>
      <p />
    </div>
  </div>
</DocumentFragment>
`;
